### DISCUSSION TITLE
need to give one button in toolbar and on click of button can we open excalidraw to 100% full blown view

### DISCUSSION THREAD

**Discussion by sohailamjad256** [excalidraw/excalidraw#4771] [Q&A] [answered]
Hi @ad1992 need to give one button in toolbar and on click of button can we open excalidraw to 100% full blown view...Thanks

**Comment by sohailamjad256** [comment]
Hii..i just need the way to achieve 100% view  on click of any button

**Reply by sohailamjad256** [reply]
HI..can i get any updates on this

**Comment by ad1992** [comment, marked as answer]
Whats the issue you are facing in implmenting this ?
You can use [renderTopRightUI](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#renderTopRightUI), [renderFooter](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#renderfooter) to add custom UI on top of excalidraw.
Please go through the documentation for the usage.

**Reply by sohailamjad256** [reply]
i just want to make full 100% view onclick of any of the icons

**Comment by sohailamjad256** [comment]
HI @ad1992 ..its little bit priority can you pls help me out ...can we give button on toolbar to increase the width and height of excalidraw and as well as decrease the height and width

**Reply by ad1992** [reply]
Where do you want to put the button? I am not really sure where you are blocked, can you take a look at the above comment and implement it with the help of the render props? In case it doesn't work, you can share code sandbox link with your implementation so we can take a look

**Reply by sohailamjad256** [reply]
actually i did'nt get the render props things ..and like we have these buttons zen mode and view mode ..there only i want one button to maximize and minimize the canvas

**Reply by ad1992** [reply]
Can you elaborate on what's not clear? In the docs, In the [code sandbox demo,](https://codesandbox.io/s/excalidraw-ehlz3?file=/src/App.js:6768-6774) we have examples as well for both the render props so I am not sure where you are blocked!

**Reply by sohailamjad256** [reply]
i don't see anywhere we are maximizing the screen to full blown view

**Comment by sohailamjad256** [comment]
or any examples of render props ..pls share i will understand from this..like sandbox link

**Reply by ad1992** [reply]
You will have to implement it yourself using the render props similar to how in the example we are showing `custom footer` / `click` button on the top right using render props
![renderprops](https://user-images.githubusercontent.com/11256141/154946107-11b55d7f-e679-4ebb-997b-5b77bcda0075.png)

**Reply by sohailamjad256** [reply]
so, with the help of render props i can implement that full blown view??

**Reply by ad1992** [reply]
of course, you can do any additional custom implementations with render props, Please go through the docs/examples, that will save time :)

**Reply by sohailamjad256** [reply]
ok.. i will try from my side to implement. If iam not able to do please help me
